From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?ZX=E5=A4=8F=E5=A4=9C=E4=B9=8B=E9=A3=8E?=
 <snwcreations@qq.com>
Date: Tue, 17 Sep 2024 13:06:33 +0800
Subject: [PATCH] build: POM changes

This commit make the project only compiles with v1_18_R2 module, starts the patch to upstream.

diff --git a/dist/pom.xml b/dist/pom.xml
index edbc40f70f3f4d2de00e0a3cc553d13539f3858c..430ae3cafe803e21a69e4eb5bcc911d1f958d84a 100644
--- a/dist/pom.xml
+++ b/dist/pom.xml
@@ -15,7 +15,10 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-assembly-plugin</artifactId>
                 <configuration>
-                    <finalName>Citizens-${CITIZENS_VERSION}-b${BUILD_NUMBER}</finalName>
+                    <!-- Backport start - We only need 1.18.2 -->
+<!--                    <finalName>Citizens-${CITIZENS_VERSION}-b${BUILD_NUMBER}</finalName>-->
+                    <finalName>Citizens-MC1.18.2-${CITIZENS_VERSION}-b${BUILD_NUMBER}</finalName>
+                    <!-- Backport end -->
                     <appendAssemblyId>false</appendAssemblyId>
                 </configuration>
                 <version>${maven-assembly-plugin.version}</version>
@@ -67,7 +70,18 @@
                     <version>${project.version}</version>
                     <type>jar</type>
                     <scope>compile</scope>
-                </dependency> 
+                </dependency>
+                <!---
+                // Backport start - We only need 1.18.2
+                -->
+                <dependency>
+                    <groupId>net.citizensnpcs</groupId>
+                    <artifactId>citizens-v1_18_R2</artifactId>
+                    <version>${project.version}</version>
+                    <type>jar</type>
+                    <scope>compile</scope>
+                </dependency>
+                <!--
                 <dependency>
                     <groupId>${project.groupId}</groupId>
                     <artifactId>citizens-v1_19_R3</artifactId>
@@ -89,8 +103,12 @@
                     <type>jar</type>
                     <scope>compile</scope>
                 </dependency>
+                // Backport end
+                -->
             </dependencies>
         </profile>
+        <!---
+        // Backport start - We only need 1.18.2
         <profile>
             <id>full</id>
             <dependencies>
@@ -213,5 +231,7 @@
                 </dependency>
             </dependencies>
         </profile>
+        // Backport end
+        -->
     </profiles>
 </project>
\ No newline at end of file
diff --git a/pom.xml b/pom.xml
index 4bc373c717e5844e1bc2dd6e3e72ac179b384c55..c8ec86e2b7b4e6b9ebd7129d3025c35545b79673 100644
--- a/pom.xml
+++ b/pom.xml
@@ -29,6 +29,20 @@
             <id>citizens-repo</id>
             <url>https://maven.citizensnpcs.co/repo</url>
         </repository>
+        <!-- Backport start - Add repositories for BuildTools-independent compilation -->
+        <repository>
+            <id>codemc-nms</id>
+            <url>https://repo.codemc.io/repository/nms/</url>
+        </repository>
+        <repository>
+            <id>minecraft</id>
+            <url>https://libraries.minecraft.net/</url>
+        </repository>
+        <repository>
+            <id>central</id>
+            <url>https://repo1.maven.org/maven2/</url>
+        </repository>
+        <!-- Backport end -->
     </repositories>
     <build>
         <defaultGoal>clean package install</defaultGoal>
@@ -41,12 +55,18 @@
             </activation>
             <modules>
                 <module>main</module> 
+                <!-- Backport start - We only need 1.18.2
                 <module>v1_19_R3</module>
                 <module>v1_20_R4</module>
                 <module>v1_21_R4</module>
+                -->
+                <module>v1_18_R2</module>
+                <!-- Backport end -->
                 <module>dist</module>
             </modules>
         </profile>
+        <!--
+        // Backport start - We only need 1.18.2
         <profile>
             <id>full</id> 
             <modules>
@@ -75,5 +95,7 @@
                 <module>dist</module>
             </modules>
         </profile>
+        // Backport end
+        -->
     </profiles>
 </project>
\ No newline at end of file
